[MRG] propagate `verbose` in `write_raw_bids` to `utils._read_events`

kalenkovich opened the pull request:
PR Description
--------------

The parameter `verbose` of `write_raw_bids` was not passed to `utils._read_raw_bids` which didn't have the `verbose` parameter at all. Consequently, `utils._read_raw_bids` did not pass the non-existent parameter to `read_events`, `find_events`, and `events_from_annotations`.

What has been done:

- added `verbose` parameter to `utils._read_raw_bids`,
- passed `verbose` in `write_raw_bids` to `utils._read_raw_bids`
- passed `verbose` in `utils._read_raw_bids` to `read_events`, `find_events`, and `events_from_annotations`

closes #448 

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [x] All comments resolved
- [x] This is not your own PR
- [x] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [x] PR description includes phrase "closes <#issue-number>"
- [x] Commit history does not contain any merge commits

agramfort commented on mne_bids/utils.py:
```diff
@@ -726,6 +726,7 @@ def _read_events(events_data, event_id, raw, ext):
         The data as MNE-Python Raw object.
     ext : str
         The extension of the original data file.
+    %(verbose)s
```
this will not work as you don't have a fill_doc decorator on the function _read_events

kalenkovich replied:
Added `@verbose` decorator. Not sure if that is the right way: just copied the decorator from `read_events`, `find_events`, and `events_from_annotations`.

jasmainak replied:
Just document it manually. You might need the function `fill_docs` to make it work.

kalenkovich replied:
Documented manually. Copied the text from `mne.utils.docs.docdict['verbose']`.

agramfort commented on mne_bids/utils.py:
```diff
@@ -747,9 +748,9 @@ def _read_events(events_data, event_id, raw, ext):
         events = events_data
     elif 'stim' in raw:
-        events = find_events(raw, min_duration=0.001, initial_event=True)
+        events = find_events(raw, min_duration=0.001, initial_event=True, verbose=verbose)
```
check flake8 errors. I suspect this line is too long

kalenkovich replied:
You are absolutely correct, This and two more lines were too long - fixed.

jasmainak commented:
@kalenkovich can you update `whats_new.rst` when you're done?

and please refer to the checklist. You need to link to the issue that you are closing

kalenkovich commented:
> @kalenkovich can you update `whats_new.rst` when you're done?

I can't find this file in `mne_bids`.

> and please refer to the checklist. You need to link to the issue that you are closing

I didn't think to look at it because it was addressed to the maintainer.

I changed the title to start with "[MRG]" and added "closes #448" to the description.

jasmainak commented:
> I can't find this file in mne_bids.

here you go: https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst. It's also linked in the checklist above!

> I didn't think to look at it because it was addressed to the maintainer.

oops :) how can we improve the wording?

kalenkovich commented:
Apologies, @jasmainak - found "whats_new.rst". Github search betrayed me.

kalenkovich commented:
> oops :) how can we improve the wording?

I would add a similar checklist to "contributing.md" and refer to it in the pull request template.

jasmainak commented:
humm but that would defeat the purpose of a checklist. I use it every time to make sure everything is in order before merging. I do expect the contributor to look at it too. I guess we can leave it like this for now. We'll change it if there are more complaints :)

agramfort commented:
@kalenkovich see that CIs are not happy.

you can see the circleci docs-build artifact details to check if you're what's new entry looks correct?

kalenkovich commented:
> @kalenkovich see that CIs are not happy.
> 
> you can see the circleci docs-build artifact details to check if you're what's new entry looks correct?

I did not add myself to the "authors" list - fixed now.

jasmainak commented on doc/whats_new.rst:
```diff
@@ -30,6 +30,7 @@ Changelog
 - :func:`mne_bids.write_raw_bids` now writes CoordinateSystemDescription as specified in BIDS Specification if CoordinateSystem is MNE-compatible, by `Adam Li`_ (`#416 <https://github.com/mne-tools/mne-bids/pull/416>`_)
 - :func:`mne_bids.write_raw_bids` and :func:`mne_bids.read_raw_bids` now handle scalp EEG if Captrak coordinate system and NAS/LPA/RPA landmarks are present, by `Adam Li`_ (`#416 <https://github.com/mne-tools/mne-bids/pull/416>`_)
 - :func:`mne_bids.get_matched_empty_room` now implements an algorithm for discovering empty-room recordings that do not have the recording date set as their session, by `Richard Höchenberger`_ (`#421 <https://github.com/mne-tools/mne-bids/pull/421>`_)
+- :func:`mne_bids.write_raw_bids` now applies `verbose` to the functions that read events, by `Evgenii Kalenkovich`_ (`#453 <https://github.com/mne-tools/mne-bids/pull/453>`_)
```
I believe this should go in the Bug section of whats_new

kalenkovich replied:
Moved it to Bug section.

sappelhoff commented on mne_bids/utils.py:
```diff
@@ -726,6 +726,9 @@ def _read_events(events_data, event_id, raw, ext):
     ext : str
         The extension of the original data file.
+    verbose : bool | str | int | None
+        If not None, override default verbose level (see :func:`mne.verbose`
+        and :ref:`Logging documentation <tut_logging>` for more).
```
```suggestion
        If not None, override default verbose level (see :func:`mne.verbose`).
```

the sphinx ref to the mne docs won't work in mne-bids, so I suggest to remove it.

agramfort replied:
+1

agramfort commented:
CIs complain https://github.com/mne-tools/mne-bids/pull/453/checks?check_run_id=793001258

agramfort commented:
@sappelhoff merge if ok for you too. thx @kalenkovich
